Add Troubleshooting section to Code Coverage docs (#30964)
DataDog/documentation: content/en/code_coverage/setup.md

### Uploading coverage reports

<div class="alert alert-info">

> There is no need to merge coverage reports before uploading them to Datadog. All reports for the same commit are aggregated automatically in the Datadog backend.

Reviewer: ```suggestion
Datadog automatically aggregates all reports for the same commit on the backend. You don't need to merge coverage reports before uploading them.
```

The `datadog-ci coverage upload` command automatically detects the format of the coverage report files based on their content and file extension.
If the command fails with the "Invalid coverage report file [...]: format could not be detected" error, try setting the format explicitly using the `--format` option, like this:

{{< code-block lang="shell" >}}
datadog-ci coverage upload --format=cobertura reports/cobertura.xml
{{< /code-block >}}

### Coverage upload outputs "Could not sync git metadata" error

Git metadata upload is only required when directly integrating your CI provider with Datadog is not possible.
If you are using a [source code provider integration][12], such as Datadog GitHub app or Gitlab integration, you can disable the git metadata upload by passing the `--skip-git-metadata-upload=1` flag to the `datadog-ci coverage upload` command, like this:

{{< code-block lang="shell" >}}
datadog-ci coverage upload --skip-git-metadata-upload=1 .
{{< /code-block >}}

### Datadog UI shows different coverage percentage values than the ones in my coverage reports

Datadog automatically merges coverage reports for the same commit.

> As a result, the coverage percentage values displayed in the Datadog UI may differ from those shown in your individual coverage reports,
> especially if there are overlapping or duplicate source code file entries.

Reviewer: ```suggestion
As a result, the coverage percentage displayed in the Datadog UI may differ from the values in your individual coverage reports, especially if those reports contain overlapping or duplicate source code file entries.
```

If you use an external tool (such as [ReportGenerator](https://reportgenerator.io/)) to merge coverage reports before uploading to Datadog,

> ensure your merged report does not contain duplicate source code file entries.

Reviewer: ```suggestion
ensure your merged reports do not contain duplicate source code file entries.
```

> Datadog will deduplicate any duplicates found, which can lead to differences in coverage values between your merged report and the values displayed in the Datadog UI.

Reviewer: ```suggestion
Datadog deduplicates overlapping files across reports, which can result in differences between your original coverage values and the merged values displayed in the Datadog UI.
```
